compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2232,6 +2232,50 @@ class Typer extends Namer
     case _                  => false
   }
 
+  /** Try to rename `tpt` to a type `T` and typecheck `new T` with given expected type `pt`.
+   */
+
+  def tryNew[T >: Untyped <: Type]
+    (treesInst: Instance[T])(tree: Trees.Tree[T], pt: Type, fallBack: => Tree)(implicit ctx: Context): Tree = {
+
+    def tryWithType(tpt: untpd.Tree): Tree =
+      tryEither { implicit ctx =>
+        val tycon = typed(tpt)
+        if (ctx.reporter.hasErrors)
+          EmptyTree // signal that we should return the error in fallBack
+        else {
+          def recur(tpt: Tree, pt: Type): Tree = pt.revealIgnored match {
+            case PolyProto(targs, pt1) if !targs.exists(_.isInstanceOf[NamedArg]) =>
```

Comment:
Add a comment explaining that NamedArg don't count since they cannot be used in new expressions.